core/ops.rs

```diff
@@ -183,7 +183,7 @@ pub struct OpState {
   pub get_error_class_fn: GetErrorClassFn,
   pub tracker: OpsTracker,
   pub last_fast_op_error: Option<AnyError>,
-  gotham_state: GothamState,
+  pub(crate) gotham_state: GothamState,
```

Comment:
`OpState` already derefs to this field. Why is it necessary to explicitly drop it?

Reply:
This was causing a crash when the inspector was being held in `OpState`. Note that I added a test to demonstrate why it's necessary (see the `inspector` test at the bottom).

I added some comments and re-organized the code to make it clearer why this is necessary. Ideally we can get rid of this manual clearing later on as we continue to refactor.